packs_list.php

```diff
@@ -233,6 +295,256 @@
+				foreach ( [ 'rando-field', 'rando-date-field', 'bogus-field', 'diag-field' ]
+				          as $minimFieldSetting )
+				{
+					$minimField = $minimModule->getProjectSetting( $minimFieldSetting );
+					if ( $minimField != '' )
+					{
+						$listMinimFields[ $minimFieldSetting ] = $minimField;
+					}
+				}
+			}
+			$listPackFields = array_values( $listMinimFields );
+			// Get the fields populated by the pack management module.
+			foreach ( [ 'packfield', 'datefield', 'countfield', 'valuefield' ] as $packField )
+			{
+				if ( $infoCategory[ $packField ] != '' )
+				{
+					$listPackFields[] = $infoCategory[ $packField ];
+				}
+			}
+			foreach( $infoCategory['extrafields'] as $packField )
+			{
+				if ( $packField['field'] != '' )
+				{
+					$listPackFields[] = $packField['field'];
+				}
+			}
+			$module->dbGetLock();
+			// Assigning or unassinging single pack, or exchanging 1 assigned and 1 unassigned pack.
+			if ( count( $listChosen ) == 1 || count( $listUnassignedPacks ) == 1 )
+			{
+				// Determine the pack's assigned record if applicable.
+				if ( count( $listAssignedPacks ) == 1 )
+				{
+					$infoPackAssignment =
+						$module->getPackAssignedRecord( $infoCategory['id'],
+						                                $listAssignedPacks[0]['id'] );
+				}
+				else
+				{
+					$infoPackAssignment = [ 'record' => $_POST['record_id'],
+					                        'event' => $_POST['event_id'],
+					                        'instance' => $_POST['instance'] ];
+				}
+				$infoData = true;
+				// Assign the unassigned pack.
+				if ( count( $listUnassignedPacks ) == 1 )
+				{
+					// Get a pack as if it was being assigned normally, but using the specific pack
+					// ID and allowing expired packs to be chosen.
+					$infoData = $infoPackAssignment === false ? false :
+					            $module->choosePack( $infoCategory['id'],
+					                                 $infoPackAssignment['record'], null,
+					                                 null, $listUnassignedPacks[0]['id'], true );
+					if ( $infoData !== false )
+					{
+						// If this is a pack exchange, do not update the date and count fields if
+						// these are applicable. If the pack has a value field then this must be
+						// updated on the record.
+						if ( count( $listAssignedPacks ) == 1 )
+						{
+							if ( $infoCategory['datefield'] != '' )
+							{
+								unset( $infoData[ $infoCategory['datefield'] ] );
+							}
+							if ( $infoCategory['countfield'] != '' )
+							{
+								unset( $infoData[ $infoCategory['countfield'] ] );
+							}
+							if ( $infoCategory['valuefield'] != '' )
+							{
+								$infoData[ $infoCategory['valuefield'] ] =
+									$listUnassignedPacks[0]['value'];
+							}
+							elseif ( isset( $listMinimFields['rando-field'] ) )
+							{
+								$infoData[ $listMinimFields['rando-field'] ] =
+									$listUnassignedPacks[0]['value'];
+							}
+						}
+						// Update the record.
```

Comment:
For a pack exchange, I think it would be a good idea to have a pop up dialogue to confirm/cancel, just in case someone had selected 2 packs by accident - after all, the pack list is long and will not all be on screen at the same time.

Reply:
It does this.

Reply:
Additionally the *Save* button becomes an *Exchange* button when 2 packs which can be exchanged are selected.